indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Integer.kt

```diff
@@ -0,0 +1,365 @@
+        val diff = longer.words.size - shorter.words.size
+        return VarUInt(MutableList<UByte>(longer.words.size) {
+            if (it >= diff) shorter.words[it - diff] or longer.words[it]
+            else longer.words[it]
+        }).apply { trim() }
+    }
+
+    infix fun xor(other: VarUInt): VarUInt {
+        val (shorter, longer) = if (other.words.size < words.size) other to this else this to other
+        val diff = longer.words.size - shorter.words.size
+        return VarUInt(MutableList<UByte>(longer.words.size) {
+            if (it >= diff) shorter.words[it - diff] xor longer.words[it]
+            else longer.words[it] xor 0u
+        }).apply { trim() }
+    }
+
+    infix fun shl(offset: Int): VarUInt {
+        //we us eit only internally require(offset >= 0) { "offset must be non-negative: $offset" }
+        if (offset == 0) return VarUInt(words.toMutableList())
+        val byteOffset = offset / 8
+
+        val bitOffset = offset % 8
+        val result = MutableList<UByte>(words.size + 1) { 0u }
+        result.indices.drop(1).forEach { index ->
+            val tmp = words[index - 1].toInt() shl bitOffset
+            val tmpH = (tmp ushr 8).toUByte()
+            val tmpL = tmp.toUByte()
+            result[index - 1] = result[index - 1] or tmpH
+            result[index] = tmpL
+        }
+        return VarUInt(result.apply { repeat(byteOffset) { add(0u) } })
+    }
+
+    infix fun shr(offset: Int): VarUInt {
+        //we use it only internally require(offset >= 0) { "offset must be non-negative: $offset" }
+        if (offset == 0) return VarUInt(words.toMutableList())
+        val byteOffset = offset / 8
+        if (byteOffset >= words.size) return VarUInt()
+
+        val bitOffset = offset % 8
+        val dropped = words.dropLast(byteOffset).toMutableList()
+
+        val result = MutableList<UByte>(dropped.size) { 0u }
+        result[result.lastIndex] = (dropped.last().toInt() ushr bitOffset).toUByte()
+        for (index in result.lastIndex - 1 downTo 0) {
+            val tmp = dropped[index].toInt() shl (8 - bitOffset)
+            result[index] = tmp.ushr(8).toUByte()
+            result[index + 1] = result[index + 1] or tmp.toUByte()
+        }
+        return VarUInt(result)
+    }
+
+    fun toAsn1VarInt(): ByteArray = throughBuffer { it.writeAsn1VarInt(this) }
+
+    fun isZero(): Boolean = (words.size == 1 && words.first() == 0.toUByte())
+
+    fun bitLength(): Int {
+        var result = words.size * 8
+        for (i in 7 downTo 0) {
+            if (words.first().toInt() and (1 shl i) == 0)
+                result--
+            else return result
+        }
+        return result
+    }
+
+    fun inv(): VarUInt = VarUInt(MutableList(words.size) { words[it].inv() })
+
+
+    operator fun compareTo(byte: UByte): Int = if (words.size > 1) 1 else words.last().compareTo(byte)
+
+
+    fun shortValue(): Int =
```

Comment:
offer `strict = true` or similar that errors on oob

Reply:
we use it internally at one point, so I made it always strict